libvast/native-plugins/feather.cpp

```diff
@@ -172,8 +172,11 @@ class active_feather_store final : public active_store {
   }
 
   [[nodiscard]] detail::generator<table_slice> slices() const override {
-    for (const auto& slice : slices_)
-      co_yield slice;
+    // We need to make a copy of the slices here because the slices_ vector may
+    // get invalidated while we iterate over it.
```

Comment:
This is possible because we iterate over the generator across multiple actor handlers because we want queries to be interruptible after every handled slice. There's no need for a mutex because it's only concurrency and not parallelism, but we cannot iterate over it while also adding to it.

Reply:
I double-checked this with @tobim, who agreed that the change is sound.